lib/BloqadeQMC/src/ising/Rydberg.jl

```diff
@@ -1,23 +1,23 @@
 using Base.Iterators
 
-
 abstract type AbstractRydberg{O <: AbstractOperatorSampler} <: AbstractLTFIM{O} end
 
-struct Rydberg{O,M <: UpperTriangular{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, L <: Lattice} <: AbstractRydberg{O}
+struct Rydberg{O,M <: AbstractMatrix{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, A} <: AbstractRydberg{O}
     op_sampler::O
     V::M
     Ω::UΩ
     δ::Uδ
-    lattice::L
+    atoms::A
```

Comment:
what is `A`? either put a type constraint or put a comment here

Reply:
I've put a type contraint :+1:

Reply:
Actually, that ended up creating a typing rabbit hole. I've left a comment of now and hope to come back to it :+1: